wordpress-mobile/WordPress-iOS
# [Stats Refresh] Queries Cancelled

Fixes #11742 
Refs. #10840

This PR improves the way the Stats views are loaded, avoiding queries being cancelled.
The main change I adopted is using a _UIPageViewController_ (with no datasource and delegate) as container. This avoids to load multiple times the period queries at the beginning. Having 2 containers allocated made the _PeriodStore_ to fetch multiple times the same queries and actions, which generated the cancel of the current fetch.

As I mentioned before the _UIPageViewController_ doesn't have delegate and datasource. This because I don't want any gesture applied and I want to set the current visible view controller only in 2 specific case:
- If the selected period is different (between insights and periods)
- If the _pageViewController_ is empty

@ScoutHarris @stevebaranski I like if you both can have a look to this 😊

## To test:
As @ScoutHarris suggested:
• The last view selected is a Period (so when the app launches and stats is first accessed, it's on the Period view).
• From a new branch, run the app in the same simulator
• When the app launches, it goes directly to stats Period
• You should load the Stats with no problem

**Important:**
This doesn't fix the problem we still have when a Stats detail is pushed. I'm working separately to improve the PeriodStore loading.

Update release notes:
